src/gbm/gbtree.cc

```diff
@@ -47,6 +49,38 @@ struct GBTreeTrainParam : public dmlc::Parameter<GBTreeTrainParam> {
   }
 };
 
+/*! \brief training parameters */
+struct DartTrainParam : public dmlc::Parameter<DartTrainParam> {
+  /*! \brief whether to not print info during training */
+  int silent;
+  /*! \brief type of sampling algorithm */
+  int samp_drop;
+  /*! \brief type of normalization algorithm */
+  int norm_drop;
+  /*! \brief how many trees are dropped */
+  float rate_drop;
+  /*! \brief whether to drop trees */
+  float skip_drop;
+  /*! \brief learning step size for a time */
+  float learning_rate;
+  // declare parameters
+  DMLC_DECLARE_PARAMETER(DartTrainParam) {
+    DMLC_DECLARE_FIELD(silent).set_default(0)
+        .describe("Not print information during trainig.");
+    DMLC_DECLARE_FIELD(samp_drop).set_default(0)
```

Comment:
use string instead of type for enumeration. see https://github.com/dmlc/xgboost/blob/master/src/tree/param.h#L74

Internally, the type is still integer, but the enumeration declaration will dispatch the strings into integers, so user can specify string directly